packages/react-components/src/components/bar-chart/barChart.tsx

```diff
@@ -56,9 +59,33 @@ export const BarChart = (props: BarChartProps) => {
     ...rest
   } = props;
 
+  const {
+    viewport: injectedViewport,
+    setViewport,
+    group,
+    lastUpdatedBy,
+  } = useViewport();
+  const viewport = passedInViewport || injectedViewport || DEFAULT_VIEWPORT;
```

Comment:
What was the reason to update the viewport logic here?

Reply:
this is the same logic used inside useTimeSeriesData to select what viewport to use, but now that i think about it we should use the [utilizedViewport](https://github.com/awslabs/iot-app-kit/pull/3003/files/3d4be9a199498f310f1852fd331f88672aabbc92#diff-a635ab59949bd78e56473e361f379b8f7a3f9ec85d098ad7a75c2edb221a0eacR111) from this same component instead